features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Create/WSDL/Steps/WSDLValidation.js

```diff
@@ -0,0 +1,121 @@
+        let promised_validation;
+        let wsdlBean = {};
+        if (uploadMethod === 'file') {
+            wsdlBean.file = file;
+            promised_validation = new_api.validateWSDLFile(file);
+        } else {
+            wsdlBean.url = url;
+            promised_validation = new_api.validateWSDLUrl(url);
+        }
+        promised_validation.then(response => {
+            const {isValid, wsdlInfo} = response.obj;
+            wsdlBean.info = wsdlInfo;
+            this.setState({valid: isValid});
+            updateWSDLValidity(isValid, wsdlBean);
+        }).catch(error => {
+            this.setState({valid: false});
+            updateWSDLValidity(false, wsdlBean);
+            console.error(error);
+            Alert.error("Error while validating WSDL!!")
+        });
+
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const {validate, updateWSDLValidity} = nextProps;
+        if (validate) {
+            this.validateWSDL(updateWSDLValidity);
+        }
+    }
+
+    render() {
+        const {uploadMethod, file, valid, url} = this.state;
+        const currentFile = Object.keys(file).length === 0 ? [] : [file];
+        const urlError = (valid === false) && (uploadMethod === 'url') ? true : (valid === true) && false; // Because of null case, which means validation haven't done yet
+        return (
+            <div>
+                <Grid item xs={10}>
+                    <Button data-type='file' raised disabled={uploadMethod === 'file'}
+                            onClick={this.toggleType}>
+                        WSDL File
+                    </Button>
+                    <Button data-type='url' raised disabled={uploadMethod === 'url'} onClick={this.toggleType}>
+                        WSDL URL
+                    </Button>
+                    <Grid container spacing={0} justify="center">
+                        <Grid item style={{minHeight: '250px'}}>
+                            {uploadMethod === 'file' ? (
+                                <FileUpload currentFiles={currentFile} onDropHandler={this.handleUploadFile}/>
+                            ) : (
+                                <form>
+                                    <FormControl error={urlError} aria-describedby="url-text">
+                                        <InputLabel htmlFor="url">WSDL URL</InputLabel>
+                                        <Input id="url" value={url} onChange={this.updateURL}/>
+                                        {valid !== null && !urlError && <Done/>}
+                                        <FormHelperText id="url-text">WSDL will be validate upon submit</FormHelperText>
+                                    </FormControl>
+                                </form>
+                            )}
+                        </Grid>
+                    </Grid>
+                </Grid>
+            </div>
+        );
+    }
+}
+
+export default withStyles(styles)(WSDLValidation)
```

Comment:
Are we going to continue withStyles ?

Reply:
Thanks for pointing out @dakshika 
No, I will remove `withstyles` HOC